src/core/border_router/routing_manager.hpp

```diff
@@ -154,36 +154,52 @@ class RoutingManager : public InstanceLocator
         kOnLinkPrefixLength     = OT_IP6_PREFIX_BITSIZE, // The length of an On-link prefix. In bits.
     };
 
+    enum : uint32_t
+    {
+        kMaxInitRtrAdvertisements = 3, // The maximum number of initial Router Advertisements.
+        kMaxRtrSolicitations = 3, // The Maximum number of Router Solicitations before sending Router Advertisements.
+    };
+
     enum : uint32_t
     {
         kDefaultOmrPrefixLifetime    = 1800u,                  // The default OMR prefix valid lifetime. In seconds.
         kDefaultOnLinkPrefixLifetime = 1800u,                  // The default on-link prefix valid lifetime. In seconds.
         kMaxRtrAdvInterval           = 600,                    // Maximum Router Advertisement Interval. In seconds.
         kMinRtrAdvInterval           = kMaxRtrAdvInterval / 3, // Minimum Router Advertisement Interval. In seconds.
-        kMaxInitRtrAdvInterval       = 16,  // Maximum Initial Router Advertisement Interval. In seconds.
-        kMaxRaDelayTime              = 500, // The maximum delay of sending RA after receiving RS. In milliseconds.
-        kRtrSolicitationInterval     = 4,   // The interval between Router Solicitations. In seconds.
-        kMaxRtrSolicitationDelay     = 1,   // The maximum delay for initial solicitation. In seconds.
-        kMaxRoutingPolicyDelay       = 1,   // The maximum delay for routing policy evaluation. In seconds.
+        kMaxInitRtrAdvInterval       = 16,   // Maximum Initial Router Advertisement Interval. In seconds.
+        kMaxRaDelayTime              = 500,  // The maximum delay of sending RA after receiving RS. In milliseconds.
+        kRtrSolicitationInterval     = 4,    // The interval between Router Solicitations. In seconds.
+        kMaxRtrSolicitationDelay     = 1,    // The maximum delay for initial solicitation. In seconds.
+        kMaxRoutingPolicyDelay       = 1,    // The maximum delay for routing policy evaluation. In seconds.
+        kRtrAdvStaleTime             = 1500, // The STALE time of Router Advertisement message and prefixes. In seconds.
```

Comment:
How did you pick 1500?

Reply:
- It is shorter than the `kDefaultOnLinkPrefixLifetime` and `kDefaultOmrPrefixLifetime` which is 1800 seconds.
- It is larger than twice of the `kMaxRtrAdvInterval` so that a BR get two chances of receiving non-solicited RAs before considering a prefix is `STALE`. I think the `STALE` time should be at least larger than `kMaxRtrAdvInterval`.

Reply:
added comments and s static assertion for it.

Reply:
RFC 4861 allows MaxRtrAdvInterval to be up to 1800 seconds.

Reply:
Thanks, what about below configuration?
```C++
kDefaultOnLinkPrefixLifetime = 2400,
kMaxRtrAdvInterval = 600,
kRtrAdvStaleTime = 2000,
```

Reply:
Having both `kDefaultOnLinkPrefixLifetime` and `kRtrAdvStaleTime` set to 1800 may work. RAs are sent with a random delay that has max value of MaxRtrAdvInterval. Thoughts?

Reply:
also works for me, updated.